internal/stack/aggregate.go

```diff
@@ -0,0 +1,55 @@
+package stack
+
+import (
+	"sort"
+
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/stringset"
+)
+
+// Aggregate aggregates two set of stacks into a merged set of stacks where compatibility exists.
+//
+// Examples:
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA", "mixinB", "run:mixinC"]}}]
+// 	stacksB = [{ID: "stack1", mixins: ["build:mixinA", "run:mixinC"]}}]
+// 	result = [{ID: "stack1", mixins: ["build:mixinA", "mixinB", "run:mixinC"]}}]
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+// 	stacksB = [{ID: "stack1", mixins: ["run:mixinC"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+// 	result = [{ID: "stack1", mixins: ["build:mixinA", "run:mixinC"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+// 	stacksB = [{ID: "stack2", mixins: ["mixinA", "run:mixinB"]}}]
+// 	result = [{ID: "stack2", mixins: ["mixinA", "run:mixinB"]}}]
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA"]}}]
+// 	stacksB = [{ID: "stack2", mixins: ["mixinA", "run:mixinB"]}}]
+// 	result = []
+//
+func Aggregate(stacksA []dist.Stack, stacksB []dist.Stack) []dist.Stack {
```

Comment:
Maybe rename to something like `MergeCompatibleStacks`? I wonder if it would also be good to rename the variables to shed light on what is compatible with what. For instance:
`stacksA` => `compatStacks`
`stacksB` => `buildpackStacks` or `newStacks`

or something similar

Reply:
Does this need to be exported?

Reply:
This is used in `internal/buildpackage`, so I think it does need to be exported.

Naming wise, the intent is clear to me personally, especially given that the package is already called `stack`. `stack.MergeCompatibleStacks` could be considered a naming stutter.

Reply:
That makes sense. I missed the package name. Perhaps `MergeCompatible` then?

Reply:
Also, what I meant by my export comment was that I think this is buildpackage-specific logic, and it should be in that package (and unexported).